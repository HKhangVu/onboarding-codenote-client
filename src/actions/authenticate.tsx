export const LOG_IN = "LOG_IN";

export function userHasAuthenticated (val) {
    return {
      type: LOG_IN,
      isLogin: val
    }
}