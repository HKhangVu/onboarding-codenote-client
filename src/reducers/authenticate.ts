const initialState = {
  isAuthenticated: false,
};

export default function authenticate(state = initialState, action) { 
  return state;
};
