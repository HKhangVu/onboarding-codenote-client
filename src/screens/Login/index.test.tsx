import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Login from './index';
import store from '../../store';
import { userHasAuthenticated } from '../../actions/authenticate';

test('render login', () => {
  store.dispatch(userHasAuthenticated(true))
  const wrapper = mount(
    <Provider store={store}>
      <Login />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
