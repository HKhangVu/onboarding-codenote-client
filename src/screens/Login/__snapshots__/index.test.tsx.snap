// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render login 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Login)>
    <Login
      userHasAuthenticated={[Function]}
    >
      <div
        className="Login"
      >
        <form
          onSubmit={[Function]}
        >
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="email"
          >
            <div
              className="form-group form-group-lg"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                <label
                  className="control-label"
                  htmlFor="email"
                >
                  Email
                </label>
              </ControlLabel>
              <FormControl
                autoFocus={true}
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                type="email"
                value=""
              >
                <input
                  autoFocus={true}
                  className="form-control"
                  id="email"
                  onChange={[Function]}
                  type="email"
                  value=""
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="password"
          >
            <div
              className="form-group form-group-lg"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                <label
                  className="control-label"
                  htmlFor="password"
                >
                  Password
                </label>
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                type="password"
                value=""
              >
                <input
                  className="form-control"
                  id="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </FormControl>
            </div>
          </FormGroup>
          <LoaderButton
            block={true}
            bsSize="large"
            disabled={true}
            isLoading={false}
            loadingText="Logging in ..."
            text="Login"
            type="submit"
          >
            <Button
              active={false}
              block={true}
              bsClass="btn"
              bsSize="large"
              bsStyle="default"
              className="LoaderButton "
              disabled={true}
              type="submit"
            >
              <button
                className="LoaderButton  btn btn-lg btn-default btn-block"
                disabled={true}
                type="submit"
              >
                Login
              </button>
            </Button>
          </LoaderButton>
        </form>
      </div>
    </Login>
  </Connect(Login)>
</Provider>
`;
